### ISSUE TITLE
java.lang.StackOverflowError

### ISSUE THREAD

**Issue by wcc526** [secretsharing/secretsharing#2]
when parse a long formatted string

String formatted = "very long string";
Part part = PartFormats.parse(formatted);

java.lang.StackOverflowError

**Comment by esialb** [comment]
Can you provide a stack trace? I without it I can only guess where the problem is.

**Comment by esialb** [comment]
I managed to break it myself, and this issue is fixed in branch [feature/large-inputs](https://github.com/secretsharing/secretsharing/tree/feature/large-inputs).  Javadoc for the changes still needs to be written, after which it will be merged into develop.  A 1.2.0 release with support for parsing very large input strings will be sent to maven central with a few days.

**Comment by esialb** [comment]
Feature branch which fixes this issue will be merged into master momentarily, despite incomplete javadoc.  Completed javadoc will be merged in as it is available.

This should let you get this fix, if only by using the 1.2.0-SNAPSHOT version, until stable 1.2.0 is released.

```xml
<repository>
  <id>sonatype-oss-snapshots</id>
  <url>https://oss.sonatype.org/content/repositories/snapshots</url>
  <releases><enabled>false</enabled></releases>
  <snapshots><enabled>true</enabled></snapshots>
</repository>

<dependency>
  <groupId>org.mitre.secretsharing</groupId>
  <artifactId>secret-sharing-scheme</artifactId>
  <version>1.2.0-SNAPSHOT</version>
</dependency>
```

**Comment by esialb** [comment]
yikes, i just realized this hasn't been released yet.  i will get a 1.2.0 release out shortly.

**Comment by esialb** [comment]
This is finally fixed in version 1.2.0, which will hit maven central in an hour or two.
